src/vs/workbench/contrib/relauncher/browser/relauncher.contribution.ts

```diff
@@ -123,6 +117,12 @@ export class SettingsChangeRelauncher extends Disposable implements IWorkbenchCo
 			}
 		}
 
+		// Profiles
+		if (typeof config.workbench?.experimental?.settingsProfiles?.enabled === 'boolean' && config.workbench.experimental.settingsProfiles.enabled !== this.settingsProfilesEnabled) {
+			this.settingsProfilesEnabled = config.workbench.experimental.settingsProfiles.enabled;
+			changed = true;
+		}
```

Comment:
```suggestion
		const newValue = typeof config.workbench?.experimental?.settingsProfiles?.enabled;
		if (typeof newValue  === 'boolean' && newValue  !== this.settingsProfilesEnabled) {
			this.settingsProfilesEnabled = newValue ;
			changed = true;
		}
```

Reply:
I think a class like this could be helpful:
```ts
class ChangeObserver<T> {
   public static create<TTypeName extends "boolean" | "string" | ...>(type: TType): ChangeObserver<TypeNameToType<TTypeName>> { ... }

   private hasValue = false;
 
   // Returns if there was a change compared to the last value
   handleChange(value: T | undefined): boolean { ... }
}
```

Usage:
```ts
private readonly windowControlsOverlayEnabled = new ChangeObserver("boolean");
private readonly settingsProfilesEnabled = new ChangeObserver("boolean");

...

if (isWindows) {
	changed |= this.windowControlsOverlayEnabled.handleChange(config?.window?.experimental?.windowControlsOverlay?.enabled);
}
```